src/transforms/pipelines/mod.rs

```diff
@@ -93,14 +96,18 @@ impl Clone for PipelineConfig {
 impl PipelineConfig {
     /// Expands a single pipeline into a series of its transforms.
     fn serial(&self) -> Box<dyn TransformConfig> {
-        let pipelines: IndexMap<String, Box<dyn TransformConfig>> = self
+        let transforms: IndexMap<String, Box<dyn TransformConfig>> = self
             .transforms
             .iter()
             .enumerate()
             .map(|(index, config)| (index.to_string(), config.clone()))
             .collect();
-
-        Box::new(expander::ExpanderConfig::serial(pipelines))
+        let transforms = Box::new(expander::ExpanderConfig::serial(transforms));
+        if let Some(ref filter) = self.filter {
```

Comment:
I think we discussed making this a required field. cc/ @binarylogic on that.

Reply:
IMO, having this optional would simplify the topology, and therefore make it faster, if no filter is used.